src/main/java/org/jenkinsci/plugins/workflow/support/pickles/ExecutorPickle.java

```diff
@@ -91,6 +92,10 @@ private ExecutorPickle(Executor executor) {
 
             @Override
             protected Executor tryResolve() throws Exception {
+                if (Jenkins.get().getInitLevel() != InitMilestone.COMPLETED) {
+                    LOGGER.fine(() -> "not going to schedule " + task + " yet because Jenkins has not yet completed startup");
+                    return null;
```

Comment:
Will any retries due to this condition count against the 5-minute or so timer before the build gets aborted automatically? Or does that only apply once the build is in the queue?

Reply:
Currently it will. You are right, it would be better to start the timer when we schedule the item.